### PR TITLE
Fix issue with accordion state, add state test to side navigation

### LINKED ISSUE #516: Accordion state doesn't update when clicking accordion title then state toggle
# Bug Report

- [x] I’ve read and understood the [Contributing guidelines](https://github.com/govau/uikit/blob/master/CONTRIBUTING.md) and have done my best effort to follow them.
- [x] I’ve read and agree to the [Code of Conduct](https://github.com/govau/uikit/blob/master/CODE_OF_CONDUCT.md).
- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

@ndavison brought this up what a legend! When clicking the title of an accordion then toggling the state, the state applies however doesn't take the current state into account. For example if the accordion is open ( by title click ) it will change the state to open when it should be changing it to closed.

There was also multiple events firing for Nathan as well, lets confirm that is not from our end.

## What I expected to happen

If the accordion is open the state should tell the accordion to close and vice versa.

## Reproducing

- Module name: accordion
- Module version: 5.0.2

Steps to reproduce:

1. Go to the [accordion react test page](https://uikit.service.gov.au/packages/accordion/tests/react/)
2. Click the title of the stateful accordion
3. Click the button `toggle state of accordion` once
4. The accordion will stay closed and not open

### PR DESCRIPTION
**Pull request by sukhrajghuman**
fixes #516 
Previously the props `onOpen` and `onClose` were not being passed into the <AUaccordion /> into the state controlled accordion example

### CODE REVIEW

**Comment by sukhrajghuman** [comment]
actually now that I think of it, I may not need to update the npm version, changelog etc. Since this is just a change to documentation, and not any source code.... hmm thoughts @adamzerella

**Comment by sukhrajghuman** [comment]
Please see #547 before reviewing this

File: packages/accordion/tests/react/index.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ class AccordionWrapper extends React.Component {
 	render() {
 		return (
 			<Fragment>
-				<AUaccordion header="State controlled accordion open" state={ this.state }>
+				<AUaccordion header="State controlled accordion open" state={ this.state } onOpen={() => {this.changeAccordion('closed')}} onClose={()=> {this.changeAccordion('closed')}}>
```
Comment (adamzerella) [inline packages/accordion/tests/react/index.js]: I think adding some spacing between the functions and parameters would spice this up a bit. ( e.g `onOpen={ () => { this.changeAccordion( 'closed' ) }`

Comment (sukhrajghuman) [reply packages/accordion/tests/react/index.js]: good pick up 😅

**Comment by alex-page** [comment]
Good work @sukhrajghuman however I reckon we should break this PR up? It is solving too much.
- PR for the state
- PR for the button change

File: packages/accordion/CHANGELOG.md [hunk 2]
```diff
@@ -42,9 +43,15 @@
...
+### v7.0.0
+
+- Added onOpen and onClose props to react state controlled accordion
+- Fixed accordion closed prop not getting updated when accordion is toggled
+
+
 ### v6.0.0
 
-- Add accordion group in a unordered list, remove `role="rab"`
+- Add accordion group in a unordered list, remove `role="tab"`
```
Comment (alex-page) [inline packages/accordion/CHANGELOG.md]: Nice pickup!

File: packages/accordion/tests/react/index.js [hunk 3]
```diff
@@ -16,18 +16,18 @@ class AccordionWrapper extends React.Component {
...
-	changeAccordion( item ) {
-		this.setState({ [ item ]: !this.state[ item ] });
+	changeAccordion() {
+		this.setState({ closed: !this.state.closed });
 	}
 
 	render() {
 		return (
 			<Fragment>
-				<AUaccordion header="State controlled accordion open" state={ this.state }>
```
Comment (alex-page) [inline packages/accordion/tests/react/index.js]: The reasoning for this, is we know users want to change state that doesn't currently exist on the accordion. I'll have a quick experimentation and see if it needs to be updated.

File: packages/accordion/tests/react/index.js [hunk 4]
```diff
@@ -16,18 +16,18 @@ class AccordionWrapper extends React.Component {
...
+				<AUaccordion header="State controlled accordion open" closed={ this.state.closed } onOpen={ () => { this.changeAccordion() }} onClose={ ()=> { this.changeAccordion() }}>
 					Some content of the accordion <a href="#url">here</a>
 				</AUaccordion>
 
-				<button type="button" onClick={ () => { this.changeAccordion( 'closed' ) } }>Toggle accordion via state</button>
+				<button type="button" onClick={ () => { this.changeAccordion() }}>Toggle accordion via state</button>
```
Comment (alex-page) [inline packages/accordion/tests/react/index.js]: This still has button code?

File: packages/accordion/src/js/module.js [hunk 5]
```diff
@@ -299,7 +299,6 @@ var AU = AU || {};
 					endSize: 0,
 					speed: speed || 250,
 					callback: function() {
-						target.style.display = 'none';
```
Comment (alex-page) [inline packages/accordion/src/js/module.js]: Lets not remove this 👍 it is for screen readers.

File: packages/accordion/src/sass/_module.scss [hunk 6]
```diff
@@ -60,6 +60,7 @@
 	font-weight: bold;
 	position: relative;
 	cursor: pointer;
+	border: none;
```
Comment (alex-page) [inline packages/accordion/src/sass/_module.scss]: @sukhrajghuman What is this for?

**Comment by alex-page** [comment]
@sukhrajghuman one last issue!

**Pull request merged**